csaf_2.1/prose/edit/src/additional-conventions.md

```diff
@@ -79,4 +79,130 @@ The `/product_tree` uses a nested structure for `branches`. Along a single path
+                      {
+                        "file_hashes": [
+                          {
+                            "algorithm": "sha256",
+                            "value": "3fb9d502d096b1dfbcdfe60eed80ddecd98c8771bf21a82bbe1752735c4dc9e2"
+                          }
+                        ],
+                        "filename": "a_4-1.bin"
+                      }
+                    ]
+                  }
+                }
+              },
+              {
+                "category": "product_version",
+                "name": "4.2",
+                "product": {
+                  "name": "Example Company Controller A Firmware 4.2",
+                  "product_id": "CSAFPID-908070603",
+                  "product_identification_helper": {
+                    "hashes": [
+                      {
+                        "file_hashes": [
+                          {
+                            "algorithm": "sha256",
+                            "value": "0a853ce2337f0608489ac596a308dc5b7b19d35a52b10bf31261586ac368b175"
+                          }
+                        ],
+                        "filename": "a_4-2.bin"
+                      }
+                    ]
+                  }
+                }
+              }
+            ],
+            "category": "product_name",
+            "name": "Controller A Firmware"
+          }
+        ],
+        "category": "vendor",
+        "name": "Example Company"
+      }
+    ],
+    "relationships": [
+      {
+        "category": "installed_on",
+        "full_product_name": {
+          "name": "Example Company Controller A Firmware 4.1 installed on Example Company Controller A 1.0",
+          "product_id": "CSAFPID-908070604"
+        },
+        "product_reference": "CSAFPID-908070602",
+        "relates_to_product_reference": "CSAFPID-908070601"
+      },
+      {
+        "category": "installed_on",
+        "full_product_name": {
+          "name": "Example Company Controller A Firmware 4.2 installed on Example Company Controller A 1.0",
+          "product_id": "CSAFPID-908070605"
+        },
+        "product_reference": "CSAFPID-908070603",
+        "relates_to_product_reference": "CSAFPID-908070601"
+      }
+    ]
+  }
+```
+
+> This requirement is important to allow for correct matching.
+> The serial number `143-D-354` identifies the `Example Company Controller A 1.0` which is in this example the hardware in its version 1.0.
+> The hash `3fb9d502d096b1dfbcdfe60eed80ddecd98c8771bf21a82bbe1752735c4dc9e2` identifies the software in the version 4.1;
+> the hash `0a853ce2337f0608489ac596a308dc5b7b19d35a52b10bf31261586ac368b175` identifies the software in the version 4.2.
+> The relationships combine the software and hardware part and form a new product which can be used in the vulnerability section to
+> assign its product status.
+>
+> A matching tool can search for the serial number in an asset database and identify the asset that has this specific hardware.
```

Comment:
Doe we intent to state "can" which to me hovers a bit too close to a MAY.

If the intent is to describe the empowerment of a "matching tool" through means of the information provided, I think we should rather formulate the statement from this angle. This would eliminate the can and maybe more clearly motivate the requirement for "appliances" to separately make hard and soft ware identifiable as well as provide an explicit relationship between the two (or relationships if there are more than two parts to be related with each other).

Or did I misunderstood the goal of the section?

Reply:
I think you got the intend of the section. It looks like you already had a wording in mind. Feel free to post it.

Reply:
Thank you! I updated it a bit to make it clearer in https://github.com/oasis-tcs/csaf/pull/857/commits/a2153c1448baef7fce77fcad820332a43f0bb6f2